graphql/resolve/query_rewriter.go

```diff
@@ -1506,6 +1701,171 @@ func addSelectionSetFrom(
 	return authQueries
 }
 
+// dqlHasAuthRules is similar to `hasAuthRules`, except it is for DQL queries.
+// If the predicate Attribute of children is not of the type `Type.Field` then
+// the corresponding child is ignored during calculation. for eg: predicates like
+// `uid`,`name@en` will be ignored.
+func dqlHasAuthRules(q *gql.GraphQuery, typ schema.Type, authRw *authRewriter) bool {
+	if q == nil || typ == nil {
+		return false
+	}
+	rn := authRw.selector(typ)
+	if rn != nil {
+		return true
+	}
+	for _, fld := range q.Children {
+		fldName := getFieldName(fld.Attr)
+		if fldName == "" {
+			continue
+		}
+		if authRules := dqlHasAuthRules(fld, typ.Field(fldName).Type(), authRw); authRules {
+			return true
+		}
+	}
+	return false
+}
+
+func getFieldName(attr string) string {
```

Comment:
May be add a TODO here or in query rewriter that dgraph directive with custom DQL auth won't work and is to be added.